### PR TITLE
WIP: Vm version check for CFR_CLASS_ACCESS_MASK

### PR DESCRIPTION
**Pull request by theresa-m**
Java 9 definition for CFR_CLASS_ACCESS_MASK includes CFR_ACC_MODULE. See https://github.com/eclipse/openj9/pull/536.

Signed-off-by: Theresa Mammarella <Theresa.T.Mammarella@ibm.com>

### CODE REVIEW

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java [hunk 1]
```diff
@@ -428,6 +429,11 @@ private static void dumpModifiers(PrintStream out, long modifiers, int modifierS
 		}
 	}
 	
+	private static boolean isJava9() {
+		String JAVA_VERSION = System.getProperty("java.version");
+		return JAVA_VERSION.startsWith("1.9.0") || JAVA_VERSION.startsWith("9");
```
Comment (pshipton) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java]: why add "1.9.0"? this isn't a supported version string.

Comment (pshipton) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java]: I don't think this is checking the correct thing, as this test checks the version of the JVM running DDR, not the version of the JVM being inspected.

File: runtime/cfdumper/main.c [hunk 2]
```diff
@@ -2330,7 +2330,7 @@ static I_32 processClassFile(J9CfrClassFile* classfile, U_32 dataLength, char* r
 			break;
 
 		case ACTION_structure:
-			dumpClassFile(classfile);
+			dumpClassFile(classfile, flags);
```
Comment (pshipton) [inline runtime/cfdumper/main.c]: These flags don't appear to be flags that contain BCT_MajorClassFileVersionMask. I traced them back to the buildFlags() function.

Comment (theresa-m) [reply runtime/cfdumper/main.c]: Its not clear to me if the buildFlags() contains the class version, however these same flags are also passed into `cfreader.c:j9bcutil_readClassFileBytes`and used to determine the vm version.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java [hunk 3]
```diff
@@ -114,6 +118,18 @@
 	 */
 	/* note that methodIndex was not used and was removed */
 	public static void j9bcutil_dumpRomMethod(PrintStream out, J9ROMMethodPointer romMethod, J9ROMClassPointer romClass, long flags, int options) throws CorruptDataException {
+		int vmVersion = getVmVersion();
+		dumpRomMethod(out, romMethod, romClass, flags, options, vmVersion);
+	}
+	
+	private static int getVmVersion() throws CorruptDataException {
+		J9JavaVMPointer vm = J9RASHelper.getVM(DataType.getJ9RASPointer());
```
Comment (theresa-m) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/J9BCUtil.java]: Is this the correct vm? I noticed this was used earlier in `DumpRomClassCommand.java`.

**Comment by DanHeidinga** [comment]
@theresa-m I see this is marked as `WIP`.  Any idea when this will be ready for further review and can have the WIP removed?

**Comment by theresa-m** [comment]
@DanHeidinga I've marked this as WIP because I'm removing CFR_CLASS_ACCESS_MASK_9 in #763. Once its approved I'll close this pr.

**Pull request closed without merging**